devenv.nix

```diff
@@ -0,0 +1,31 @@
+{ pkgs, ... }:
+
+{
+  # https://devenv.sh/basics/
+
+  # https://devenv.sh/packages/
+  # manually set rust packages rather than use devenv language support because
+  # it doesn't seem to be up to date for macos yet (link error)
+  packages = with pkgs; [ cargo rustc rust-analyzer rustfmt clippy ];
```

Comment:
```suggestion
  packages = with pkgs; [ cargo rustc rust-analyzer rustfmt clippy miri ];
```

Reply:
do we need miri for local?

Reply:
If it's for the dev environment, I'd say it's one of the normal tools to have.